### DISCUSSION TITLE
Send mail from eg. Shelly via Tasmota per http to smtp-server "SENDMAIL"?

### DISCUSSION THREAD

**Discussion by PatrickR68** [arendst/Tasmota#20360] [Q&A]
Hello all, 

I compiled Tasmota v13.3.0 Quinlan as a 
DE (german) version with SENDMAIL activated. 
Flashed it on a Shelly 1 (IP-200). 

The command 
**sendmail [mail.gmx.net:465:sender@gmx.de:PWD:<sender@gmx.de>:<reciever@gmx.de>:Subject] E-mail-text** 
works in the console.  

Now I want to send data from an other Shelly -by using http cmd- as email via the Shelly with Tasmota incl. SENDMAIL .

Is this possible, if yes, how ?

Many thanks,
Patrick

**Comment by barbudor** [comment]
There's 2 way
1) if you want the other Shelly to be able to specify all the details, you can send the complete command as described in the docs:
https://tasmota.github.io/docs/Commands/#with-web-requests

You will have to escape spaces and some other characters with '%<asciicode>'
Like space be omes '%20'

2) if you just want the other Shelly not to deal with the smtp server details etc.. You can create an event-rule that the other Shelly can trigger with an event command

**Comment by PatrickR68** [comment]
Hello @barbudor, 

many thanks for your answer. 

Option 2 seems the better solution, but at first I have to learn how to do event-rule/event command. 

Maybe you have an good link with examples ?

Many thanks,
Patrick

**Reply by barbudor** [reply]
You can create an event-rule with a trigger such as `on event#eventname do ... `
So for exemple:
```
Rule1 on event#sendmail do sendmail [mail.gmx.net:465:sender@gmx.de:PWD:sender@gmx.de:reciever@gmx.de:Subject] E-mail-text endon
Rule1 1
```
This event can be called from the console:
```
event sendmail
```
Or from the other shelly through HTTP with
```
http://ip.of.first.shelly/cm?cmnd=event%20sendmail
```

So by just calling the event, the whole sendmail command is executed

Now you may want the other shelly to specify a given message. For that you can pass a value to the event:
```
Rule1 on event#sendmail do sendmail [mail.gmx.net:465:sender@gmx.de:PWD:sender@gmx.de:reciever@gmx.de:Subject] %value% endon
```
can be called with
```
event sendmail=some message
```
or in HTTP:
```
http://ip.of.first.shelly/cm?cmnd=event%20sendmail%3Dsome%20message
```
where `%3D` is `=`

**Comment by PatrickR68** [comment]
Hello @barbudor, 
many thanks. 

Could I do .. reciever@gmx.de:%value%] %value% endon"

With .. cmnd=event%20sendmail%3Dsubject%3Dsome%20message 

Patrick

**Comment by PatrickR68** [comment]
Hello @barbudor, 

many thanks. Your code works perfect. 

Some question I still have and maybe you can help me with them?
1) use 2x data (1=subject; 2=text-message), with 1x value it works 
2) text-message sent in the email with capital letters (independent that I used eg. Hallo)

3) could I select "Shelly 1 (46)" by default in the code also the names ? 

Many thanks.
Patrick

**Reply by sfromis** [reply]
1. `Event` is only one value. But you could do an extra command and store in a var.
2. Folding to uppercase is part of how rules work
3. Not clear what you want to do there

I suppose that event with `MySubject] MyMessage` could also work if having `... reciever@gmx.de:%value%`.
Sure, it is a messy thing to do, but such stuff *can* be done.

**Comment by PatrickR68** [comment]
Hello @sfromis, 
many thanks for your answer. 

With 3) I mean to change the configuration of the module/names -as I do it manually- to have it already in the compiled FW. 
I managed this now :)

But would 1) and your proposal look like ?

Patrick

**Reply by sfromis** [reply]
1: Two Event commands, the first matching a rule like `on Event#setsubject do var1 %value% endon`

3: I think that you could `#define MODULE SHELLY1` https://github.com/arendst/Tasmota/blob/development/tasmota/include/tasmota_template.h#L1433-L1460

**Comment by PatrickR68** [comment]
Hello @sfromis, 

sorry to ask again, but I don't find it. 
How do I program this two Event commands correct ?

#define USER_RULE1 
   "on event#setsubject do var1 %value% endon  on event#shellymail do sendmail [*:*:*:*:*:<receiver@gmx.com>:var1] %value% endon"          
#define USER_BACKLOG "Rule1 1"

And in HTTP:
http://ip.of.first.shelly/cm?cmnd=event%20setsubject%3Dsubject
http://ip.of.first.shelly/cm?cmnd=event%20sendmail%3Dmessage

**Reply by sfromis** [reply]
Use `%var1%` to get it expanded.

**Comment by PatrickR68** [comment]
Hello @sfromis, 
oh yes, I didn't saw my mistake. 

#define USER_RULE1 "on event#setsubject do var1 %value% endon  on event#shellymail do sendmail [*:*:*:*:*:<receiver@gmx.com>:%var1%] %value% endon"          

#define USER_BACKLOG "Rule1 1"

in HTTP I use two commands:
http://ip.of.first.shelly/cm?cmnd=event%20setsubject%3DSubjectV7
http://ip.of.first.shelly/cm?cmnd=event%20shellymail%3DmessageV7 

4) Could I send two commands in one HTTP or do I have to send two like above ? 

5) Or to send one value (containing subject:#:message) which has to be splitted at eg. :#: first and then to do the sendmail with var1 and var2 and without additional event (so only one event to do it)

Patrick

**Reply by sfromis** [reply]
Should work with multiple commands via the Backlog command. No string handling features, like splitting, available with ESP8266 and rules.

**Comment by PatrickR68** [comment]
@sfromis 

Like this ?

http://ip.of.first.shelly/cm?cmnd=Backlog%20event%20setsubject%3DSubjectV7d%3Bevent%20shellymail%3DmessageV7d

**Reply by sfromis** [reply]
Well, see what happens when you try. Watch the console.
